client/consensus/babe/src/lib.rs

```diff
@@ -1102,24 +1114,27 @@ where
 {
 	async fn verify(
 		&mut self,
-		origin: BlockOrigin,
-		header: Block::Header,
-		justifications: Option<Justifications>,
-		mut body: Option<Vec<Block::Extrinsic>>,
+		mut block: BlockImportParams<Block, ()>,
 	) -> BlockVerificationResult<Block> {
 		trace!(
 			target: "babe",
 			"Verifying origin: {:?} header: {:?} justification(s): {:?} body: {:?}",
-			origin,
-			header,
-			justifications,
-			body,
+			block.origin,
+			block.header,
+			block.justifications,
+			block.body,
 		);
 
-		let hash = header.hash();
-		let parent_hash = *header.parent_hash();
+		let hash = block.header.hash();
+		let parent_hash = *block.header.parent_hash();
+
+		if block.with_state() {
+			// When importting whole state we don't calculate epoch descriptor, but rater
```

Comment:
```suggestion
			// When importing whole state we don't calculate epoch descriptor, but rather
```

Reply:
If I'm understanding correctly we can skip all the verifications below since we just trust this block right?

Reply:
Yes. There are no parent blocks or current epoch changes to verify against anyway.